mypy/messages.py

```diff
@@ -796,11 +796,15 @@ def signature_incompatible_with_supertype(
             name, target), context)
 
     def argument_incompatible_with_supertype(
-            self, arg_num: int, name: str, type_name: Optional[str],
-            name_in_supertype: str, supertype: str, context: Context) -> None:
+            self, arg_num: int, name: str, arg_type: Type, type_name: Optional[str],
+            name_in_supertype: str, arg_type_in_supertype: Type, supertype: str,
+            context: Context) -> None:
         target = self.override_target(name, name_in_supertype, supertype)
-        self.fail('Argument {} of "{}" incompatible with {}'
-                  .format(arg_num, name, target), context)
+        arg_type_f = self.format_bare(arg_type)
+        arg_type_in_supertype_f = self.format_bare(arg_type_in_supertype)
+        self.fail('Argument {} of "{}" of type "{}" is incompatible with {}; '
+                  'supertype defines the argument type as "{}"'
+                  .format(arg_num, name, arg_type_f, target, arg_type_in_supertype_f), context)
```

Comment:
I don't think you need `arg_type_f` in this function. The third format argument is the name of the superclass (it was already OK). You only need `arg_type_in_supertype_f`.

Reply:
Fair enough. I've updated the code 👍